src/domTree.js

```diff
@@ -143,6 +143,133 @@ class span {
 }
 
 /**
+ * This node represents an anchor (<a>) element with a hyperlink, a list of classes,
+ * a list of children, and an inline style. It also contains information about its
+ * height, depth, and maxFontSize.
+ */
+class anchor {
+    constructor(href, classes, children, options) {
+        this.href = href || "";
+        this.classes = classes || [];
+        this.children = children || [];
+        this.height = 0;
+        this.depth = 0;
+        this.maxFontSize = 0;
+        this.style = {};
+        this.attributes = {};
+        if (options) {
+            if (options.style.isTight()) {
+                this.classes.push("mtight");
+            }
+            if (options.getColor()) {
+                this.style.color = options.getColor();
+            }
+        }
+    }
+
+    /**
+     * Sets an arbitrary attribute on the anchor. Warning: use this wisely. Not all
+     * browsers support attributes the same, and having too many custom attributes
+     * is probably bad.
+     */
+    setAttribute(attribute, value) {
+        this.attributes[attribute] = value;
+    }
+
+    tryCombine(sibling) {
+        return false;
+    }
+
+    /**
+     * Convert the anchor into an HTML node
+     */
+    toNode() {
+        const a = document.createElement("a");
+
+        // Apply the href
+        a.setAttribute('href', this.href);
+
+        // Apply the class
+        if (this.classes.length) {
+            a.className = createClass(this.classes);
+        }
+
+        // Apply inline styles
+        for (const style in this.style) {
+            if (Object.prototype.hasOwnProperty.call(this.style, style)) {
+                a.style[style] = this.style[style];
+            }
+        }
+
+        // Apply attributes
+        for (const attr in this.attributes) {
+            if (Object.prototype.hasOwnProperty.call(this.attributes, attr)) {
+                a.setAttribute(attr, this.attributes[attr]);
+            }
+        }
+
+        // Append the children, also as HTML nodes
+        for (let i = 0; i < this.children.length; i++) {
+            a.appendChild(this.children[i].toNode());
+        }
+
+        return a;
+    }
+
+    /**
+     * Convert the a into an HTML markup string
+     */
+    toMarkup() {
+        let markup = "<a";
+
+        // Add the href
+        markup += " anchor=\"";
+        markup += utils.escape(this.href);
+        markup += "\"";
```

Comment:
> Since we're adding href to attributes in the constructor shouldn't the loop below that adds all other attributes also add this one?

Ah, yes, that's right.
And I mistyped `anchor` instead of `href` above, and this mistakenly prevents the duplication of `href` attribute. I must fix this.